src/indexer/utils/substrate/index.ts

```diff
@@ -0,0 +1,317 @@
+  const { destDomainId: destinationDomainId, depositNonce, txIdentifier, blockNumber, depositData, handlerResponse, sender, resourceId, timestamp
+  } = substrateDepositData
+
+  const decodedAmount = getDecodedAmount(depositData)
+
+  const foundTransfer = await transferRepository.findByNonceFromDomainId(Number(depositNonce), `${originDomainId}`)
+
+  if (foundTransfer) {
+    const dataTransferToUpdate = {
+      depositNonce: Number(depositNonce),
+      sender,
+      amount: decodedAmount,
+      resourceID: resourceId,
+      fromDomainId: `${originDomainId}`,
+      toDomainId: `${destinationDomainId}`,
+      timestamp: timestamp,
+      destination: '' // FIX
+    }
+    try {
+      await transferRepository.updateTransfer(dataTransferToUpdate, foundTransfer.id)
+    } catch (e) {
+      logger.error(`Error updating substrate deposit transfer: ${e}`)
+    }
+  }
+
+
+  const transferData = {
+    id: new ObjectId().toString(),
+    depositNonce: Number(depositNonce),
+    sender,
+    amount: decodedAmount,
+    resourceID: resourceId,
+    fromDomainId: `${originDomainId}`,
+    toDomainId: `${destinationDomainId}`,
+    timestamp: timestamp,
+  } as Pick<DecodedDepositLog, "depositNonce" | "sender" | "amount" | "resourceID" | "toDomainId" | "fromDomainId" | "timestamp">
+
+  let insertedTransfer: Transfer | undefined
+  try {
+    insertedTransfer = await transferRepository.insertSubstrateDepositTransfer(transferData)
+  } catch (e) {
+    logger.error(`Error inserting substrate deposit: ${e}`)
+  }
+
+  if (insertedTransfer !== undefined) {
+    const deposit = {
+      id: new ObjectId().toString(),
+      type: SubstrateTypeTransfer.Fungible,
+      txHash: txIdentifier,
+      blockNumber: blockNumber,
+      depositData: depositData,
+      handlerResponse: handlerResponse,
+      transferId: insertedTransfer!.id,
+    }
+
+    try {
+      await depositRepository.insertDeposit(deposit)
+    } catch (e) {
+      logger.error(`Error inserting substrate deposit: ${e}`)
+    }
+
+  }
+}
+
+function getDecodedAmount(depositData: string): string {
+  const abiCoder = AbiCoder.defaultAbiCoder()
+  const parsedAmount = `0x${depositData.substring(2).slice(0, 64)}`
+  const decodedDepositData = abiCoder.decode(["uint256"], parsedAmount)
+  return formatEther(decodedDepositData[0].toString())
+}
+
+export async function saveEvents(
+  blockHash: BlockHash,
+  provider: ApiPromise,
+  block: number,
+  domain: Domain,
+  executionRepository: ExecutionRepository,
+  transferRepository: TransferRepository,
+  depositRepository: DepositRepository,
+  domainRepository: DomainRepository,
+) {
+  const signedBlock = await provider.rpc.chain.getBlock(blockHash)
+  const at = await provider.at(blockHash)
+  const timestamp = Number((await at.query.timestamp.now()).toString())
+  const allRecords = await at.query.system.events() as unknown as Array<SubstrateEvent>
+
+  // we get the proposal execution events - ts-ignore because of allRecords
```

Comment:
Update comments to remove ts-ignore explanations

Reply:
done